include/openenclave/bits/result.h

```diff
@@ -372,6 +372,11 @@ typedef enum _oe_result
      */
     OE_QUOTE_LIBRARY_LOAD_ERROR,
 
+    /**
+     * The library API used for quote generation and attestation failed.
+     */
+    OE_QUOTE_LIBRARY_ERROR,
```

Comment:
The term "quote" is SGX specific and so the error should have SGX in it.
Either that or replace "quote" with evidence if this is intended to be a TEE-agnostic result code.

```suggestion
     * The library API used for SGX quote generation and attestation failed.
     */
    OE_SGX_QUOTE_LIBRARY_ERROR,
```

Reply:
@dthaler  "QUOTE" has been widely used in the errors. If we add "SGX" or replace with "evidence", it looks inconsistent among other already defined errors. I suggest we leave it as `OE_QUOTE_LIBRARY_ERROR` and open a separate issue to fix the naming inconsistence without breaking customer's code. For example,

```
typedef enum _oe_result
{
    ....
    OE_QUOTE_PROVIDER_LOAD_ERROR,
    OE_SGX_QUOTE_PROVIDER_LOAD_ERROR = OE_QUOTE_PROVIDER_LOAD_ERROR,

    OE_QUOTE_VERIFICATION_ERROR,
    OE_SGX_QUOTE_VERIFICATION_ERROR = OE_QUOTE_VERIFICATION_ERROR,

    OE_QUOTE_LIBRARY_LOAD_ERROR,
    OE_SGX_QUOTE_LIBRARY_LOAD_ERROR = OE_QUOTE_LIBRARY_LOAD_ERROR,

    ....
}
```

Reply:
Agree that existing errors are similarly bad.  I agree with aliasing the existing ones, though the alias should be in the other direction, i.e.,
```
    OE_SGX_QUOTE_PROVIDER_LOAD_ERROR,
    OE_QUOTE_PROVIDER_LOAD_ERROR = OE_SGX_QUOTE_PROVIDER_LOAD_ERROR,
```
This is so that the alias could for example be put inside an ifdef as a way to someday remove support.

But I would recommend not adding to the debt and doing new ones with SGX (or EVIDENCE) now so that we don't introduce new app code that would need to change in the next release.